pkg/registry/core/trace/nse_registry.go

```diff
@@ -138,24 +154,28 @@ type traceNetworkServiceEndpointRegistryServer struct {
 }
 
 func (t *traceNetworkServiceEndpointRegistryServer) Register(ctx context.Context, in *registry.NetworkServiceEndpoint) (*registry.NetworkServiceEndpoint, error) {
+	if log.Fields(ctx) != nil {
+		// don't change type if it's already present - it happens when registry elements used in endpoint discovery
+		if _, ok := log.Fields(ctx)["type"]; !ok {
+			log.Fields(ctx)["type"] = networkServiceEndpointRegistry
+		}
```

Comment:
Can we put these fields (type and id) on fields creating?

Reply:
ok, done